File: timelineview/src/main/java/com/konatsup/timelineview/CellViewHolder.java
package com.konatsup.timelineview;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

public class CellViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
    private LinearLayout mCell;
    private ImageView mImageView;
    private View mStartBar;
    private View mEndBar;
    private CellViewAdapter mAdapter;

    public CellViewHolder(View itemView, CellViewAdapter adapter) {
        super(itemView);
        itemView.setOnClickListener(this);

        mAdapter = adapter;
        mCell = (LinearLayout) itemView.findViewById(R.id.cell);
        mImageView = (ImageView) itemView.findViewById(R.id.imageView);
        mStartBar = (View) itemView.findViewById(R.id.startBar);
        mEndBar = (View) itemView.findViewById(R.id.endBar);
    }

    @Override
    public void onClick(View v) {
        mAdapter.onItemHolderClick(this);
    }

    public void setCellColor(int color) {
        mCell.setBackgroundColor(color);
    }

    public void setText(int position) {
        ((TextView) mCell.findViewById(R.id.text)).setText(String.valueOf(position));
    }

    public void setImage(int resId) {
        if (resId == -1) {
            mImageView.setImageDrawable(null);
        } else {
            mImageView.setImageResource(resId);
        }
    }

    public void setVisibility(boolean isStart, boolean isEnd) {
        if (isStart) {
            mStartBar.setVisibility(View.VISIBLE);
        } else {
            mStartBar.setVisibility(View.GONE);
        }

        if (isEnd) {
            mEndBar.setVisibility(View.VISIBLE);
        } else {
            mEndBar.setVisibility(View.GONE);
        }

    }

}

File: timelineview/src/main/java/com/konatsup/timelineview/CellViewHolder.kt
package com.konatsup.timelineview

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView

class CellViewHolder(itemView: View, private val adapter: CellViewAdapter) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
    private val cell: LinearLayout
    private val imageView: ImageView
    private val startBar: View
    private val endBar: View

    init {
        itemView.setOnClickListener(this)
        cell = itemView.findViewById<View>(R.id.cell) as LinearLayout
        imageView = itemView.findViewById<View>(R.id.imageView) as ImageView
        startBar = itemView.findViewById(R.id.startBar) as View
        endBar = itemView.findViewById(R.id.endBar) as View
    }

    override fun onClick(v: View) {
        adapter.onItemHolderClick(this)
    }

    fun setCellColor(color: Int) {
        cell.setBackgroundColor(color)
    }

    fun setText(position: Int) {
        (cell.findViewById<View>(R.id.text) as TextView).text = position.toString()
    }

    fun setImage(resId: Int) {
        if (resId == -1) {
            imageView.setImageDrawable(null)
        } else {
            imageView.setImageResource(resId)
        }
    }

    fun setVisibility(isStart: Boolean, isEnd: Boolean) {
        if (isStart) {
            startBar.visibility = View.VISIBLE
        } else {
            startBar.visibility = View.GONE
        }

        if (isEnd) {
            endBar.visibility = View.VISIBLE
        } else {
            endBar.visibility = View.GONE
        }

    }

}